pool/pool.go

```diff
@@ -1307,32 +1327,32 @@ func (p *Pool) initCallContext(ctx *callContext, cfg prmCommon, prmCtx prmContex
 func (p *Pool) openDefaultSession(ctx *callContext) error {
 	cacheKey := formCacheKey(ctx.endpoint, ctx.key)
 
-	tok := p.cache.Get(cacheKey)
-	if tok == nil {
+	tok, ok := p.cache.Get(cacheKey)
+	if !ok {
 		var err error
 		var sessionOwner user.ID
 
 		user.IDFromKey(&sessionOwner, ctx.key.PublicKey)
 
 		// open new session
-		tok, err = createSessionTokenForDuration(ctx, ctx.client, sessionOwner, p.stokenDuration)
+		t, err := createSessionTokenForDuration(ctx, ctx.client, sessionOwner, p.stokenDuration)
 		if err != nil {
 			return fmt.Errorf("session API client: %w", err)
 		}
 
 		// cache the opened session
-		p.cache.Put(cacheKey, tok)
+		p.cache.Put(cacheKey, *t)
 	}
 
-	tokToSign := *tok
-	tokToSign.SetContext(ctx.sessionContext)
+	tok.ForVerb(ctx.sessionVerb)
+	tok.ApplyTo(ctx.sessionAddr)
```

Comment:
I think you missed `tok = *t`. Otherwise `tok` will have empty lifetime.

Reply:
a66922fe096887939fcc755adf0b687935f145eb